apps/nextjs/src/components/AppComponents/Chat/chat-right-hand-side-lesson.tsx

```diff
@@ -23,7 +24,10 @@ const ChatRightHandSideLesson = ({
   closeMobileLessonPullOut,
   demo,
 }: Readonly<ChatRightHandSideLessonProps>) => {
-  const { messages } = useLessonChat();
+  // const { messages } = useLessonChat();
+  const stableMessages = useChatStore((state) => state.stableMessages);
+  const streamingMessage = useChatStore((state) => state.streamingMessage);
+  const messages = [...stableMessages, streamingMessage];
```

Comment:
As above, in the long run we don't want to do this, but for this PR we could useMemo and put in a comment